tests/test_preload_script.py

```diff
@@ -95,3 +95,290 @@ async def test_addPreloadScript_logging(websocket, context_id):
+                "resultOwnership": "root"
+            }
+        })
+    assert result["result"] == {"type": "string", "value": 'bar'}
+
+    await execute_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "url": "data:text/html,<h2>second context</h2>",
+                "wait": "complete",
+                "context": another_context_id
+            }
+        })
+
+    result = await execute_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": "window.foo",
+                "target": {
+                    "context": another_context_id
+                },
+                "awaitPromise": True,
+                "resultOwnership": "root"
+            }
+        })
+    assert result["result"] == {"type": "string", "value": 'bar'}
+
+
+# TODO: make offline.
+@pytest.mark.parametrize("nested_iframe", [
+    'about:blank',
+    'https://example.com/',
+    'data:text/html,<h2>child page</h2>',
+])
+@pytest.mark.asyncio
+async def test_addPreloadScriptGlobally_loadedInMultipleContexts_withIframes(
+        websocket, context_id, nested_iframe):
+    await execute_command(
+        websocket, {
+            "method": "script.addPreloadScript",
+            "params": {
+                "expression": "() => { window.foo='bar'; }",
+            }
+        })
+
+    await execute_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "url": "data:text/html,<h2>first context</h2>",
+                "wait": "complete",
+                "context": context_id
+            }
+        })
+
+    result = await execute_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": "window.foo",
+                "target": {
+                    "context": context_id
+                },
+                "awaitPromise": True,
+                "resultOwnership": "root"
+            }
+        })
+    assert result["result"] == {"type": "string", "value": 'bar'}
+
+    # Create a new iframe within the same context.
+    result = await execute_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": f"""const iframe = document.createElement('iframe');
+                    iframe.src = '{nested_iframe}';
+                    document.body.appendChild(iframe);""",
+                "target": {
+                    "context": context_id
+                },
+                "awaitPromise": True,
+                "resultOwnership": "root"
+            }
+        })
+
+    result = await execute_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": "window.foo",
+                "target": {
+                    "context": context_id
```

Comment:
You check the original page has `window.foo`, not the newly created iframe.

Reply:
Done.